qa-pipelines/flags.yml

```diff
@@ -4,37 +4,45 @@
 # The settings in this file will also be valid for our most complete pipeline
 # (which performs an upgrade and all tests)
 
-# deploy cf for upgrade pipelines
-enable-cf-deploy-pre-upgrade: true
+# Enable HA and non-HA pipelines
+ha: true
+sa: true
+
+# Report status as steps complete
+status-reporting: true
+
+# version to upgrade from; if null, upgrade is not performed.
+upgrade-from-version: scf-sle-2.15.2+cf3.6.0.0.gde1bd02f
+
+# Deploy CF USB before upgrading
+enable-cf-usb-deploy-pre-upgrade: true
```

Comment:
The deploy USB task and post-upgrade USB task should only be used before and after the upgrade, in upgrade pipelines; it doesn't hurt to clarify that this is a pre-upgrade task, but the post-upgrade component of it should still have `post-upgrade` in the name (in contrast with other tasks which may be run as post-upgrade tasks), because this one should *only* run after an upgrade (where the pre-upgrade USB task ran prior to the upgrade)

Reply:
After looking at this, I don't think USB deploy needs to be two separate tasks. There's a pre-upgrade and post-upgrade component we created as an integration test to reproduce and test resolution of an old bug, which I think was resolved in USB via https://github.com/SUSE/cf-usb/pull/16

The pre-upgrade usb deploy task doesn't do the same kind of testing on its own, and we've only ever used those two tasks in unison (and haven't even been running those tasks lately). The comments in the flags.yml file give a bit of clarity into the intent:

`# deploy mysql and postgres services, sidecars, and sample apps before an upgrade`

and

`# test apps which use services, and tear down apps, services, and sidecars`

respectively.

The post-upgrade part would *never* be run in a non-upgrade pipeline because the stuff it expected wouldn't exist (and the pre-upgrade task only runs before the deploy task which would run in a non-upgrade pipeline)

We could modify pipeline behaviour to run both tasks on a straight deploy, or even after an upgrade, but since we've deprioritized USB testing, it's not something we've had a card for. As it is, those two tasks were specifically designed to work in unison in upgrade pipelines as an integration test.

I do think we really should have only had one flag which enabled or disabled both of these tasks ( `enable-usb-upgrade-integration` ?)

Reply:
Okay, 1ced19a27b16313a31e5263996b5234fd5954a40 added to follow this description (which seems correct).  Called it `enable-cf-usb-upgrade-tests`. Thanks!